build.rs

```diff
@@ -52,21 +55,29 @@ fn prost_compile_dataset(name: &'static str) -> std::io::Result<()> {
     }
     let mut prost_config = prost_build::Config::new();
     prost_config.protoc_arg("--experimental_allow_proto3_optional");
+    prost_config.out_dir(format!("./src/datasets/{name}"));
     prost_config.compile_protos(
-        &[format!("./src/datasets/{0}/{0}.proto", name).as_str()],
+        &[format!("./src/datasets/{name}/{name}.proto").as_str()],
         &["src"],
     )
 }
 
 fn main() {
-    const DATASETS: &[&str] = &["log", "mesh", "minecraft_savedata", "mk48"];
-    for &name in DATASETS.iter() {
-        // bebop_compile_dataset(name);
-        #[cfg(all(feature = "capnp", not(feature = "use-committed-capnp")))]
-        capnpc_compile_dataset(name).unwrap();
-        #[cfg(all(feature = "flatbuffers", not(feature = "use-committed-flatbuffers")))]
-        flatc_compile_dataset(name).unwrap();
-        #[cfg(feature = "prost-build")]
-        prost_compile_dataset(name).unwrap();
+    #[cfg(any(
+        feature = "regenerate-capnp",
+        feature = "regenerate-flatbuffers",
+        feature = "regenerate-prost"
+    ))]
+    {
+        const DATASETS: &[&str] = &["log", "mesh", "minecraft_savedata", "mk48"];
+        for &name in DATASETS.iter() {
+            // bebop_compile_dataset(name);
```

Comment:
what's needed to enable bebop in this repo? i saw them posting publicity about their encoding, had a look at how the encoding actually works (it's just alright), and had a look at their repo (where they are making some ***WILD*** unsubstantiated claims about relative performance, [even after they claim to have addressed those discrepancies](https://github.com/betwixt-labs/bebop/issues/312). there's just no evidence that they have any interest in real numbers to back up their claims, lol

Reply:
#23 is the blocking issue, it looks like https://github.com/betwixt-labs/bebop/issues/153 is what I ran into. It's not strictly a blocking issue, but the amount of work it would add is pretty significant.

Reply:
> what's needed to enable bebop in this repo? i saw them posting publicity about their encoding, had a look at how the encoding actually works (it's just alright), and had a look at their repo (where they are making some _**WILD**_ unsubstantiated claims about relative performance, [even after they claim to have addressed those discrepancies](https://github.com/betwixt-labs/bebop/issues/312). there's just no evidence that they have any interest in real numbers to back up their claims, lol

While not as exhaustive as this project, you can find our Rust benchmarks [in the lab](https://github.com/betwixt-labs/bebop/tree/master/Laboratory/Rust/benchmarking).

Reply:
yeah it seems like bebop's rust library usability is not necessarily great in existing projects, or perhaps just has very poor ergonomics generally, and there appears to be little to no motivation to improve it from their project. i can only imagine this contributes to the dearth of real comparative benchmarks, something of a bellwether for an encoding library's quality in my experience